lib/java_buildpack/framework/app_dynamics_agent.rb

```diff
@@ -83,7 +84,7 @@ def host_name(java_opts, credentials)
 
       def node_name(java_opts, credentials)
         name = credentials['node-name'] || @configuration['default_node_name']
-        java_opts.add_system_property('appdynamics.agent.nodeName', "#{name}")
+        java_opts.add_system_property('appdynamics.agent.nodeName', @appdynamics_tier_name + ":#{name}")
```

Comment:
There are other multiple views within the controller, that only shows node names. It becomes difficult map those nodes to appropriate tiers when we use `instance_index` as node-name.  This is for better readability sake.